apix-impl/src/main/java/eu/xenit/apix/alfresco/comments/CommentService.java

```diff
@@ -0,0 +1,134 @@
+        return apixConverter.apix(commentService.getDiscussableAncestor(apixConverter.alfresco(commentNodeRef)));
+    }
+
+    @Override
+    public Conversation getComments(NodeRef targetNode, int skipCount, int pageSize) {
+        org.alfresco.service.cmr.repository.NodeRef alfTargetNode = apixConverter.alfresco(targetNode);
+        PagingResults<org.alfresco.service.cmr.repository.NodeRef> commentAlfNodes =
+                commentService.listComments(alfTargetNode, new PagingRequest(skipCount, pageSize));
+        return new Conversation()
+                .comments(commentAlfNodes.getPage().stream()
+                        .map(alfCommentNode -> toComment(alfTargetNode, alfCommentNode)).collect(Collectors.toList()))
+                .hasMore(commentAlfNodes.hasMoreItems())
+                .canCreate(canCreateComment(targetNode));
+    }
+
+    @Override
+    public Comment addNewComment(NodeRef targetNode, String content) {
+        org.alfresco.service.cmr.repository.NodeRef alfTargetNode = apixConverter.alfresco(targetNode);
+        return toComment(alfTargetNode, commentService.createComment(alfTargetNode, "", content, false));
+    }
+
+    @Override
+    public Comment updateComment(NodeRef targetCommentNode, String content) {
+        return updateComment(getTargetDocumentForComment(targetCommentNode), targetCommentNode, content);
+    }
+
+    @Override
+    public Comment updateComment(NodeRef targetDocumentNode, NodeRef targetCommentNode, String content) {
+        org.alfresco.service.cmr.repository.NodeRef alfTargetCommentNode = apixConverter.alfresco(targetCommentNode);
+        commentService.updateComment(alfTargetCommentNode, "", content);
+        org.alfresco.service.cmr.repository.NodeRef alfTargetDocumentNode = apixConverter.alfresco(targetDocumentNode);
+        return toComment(alfTargetDocumentNode, alfTargetCommentNode);
+    }
+
+    @Override
+    public void deleteComment(NodeRef targetCommentNode) {
+        commentService.deleteComment(apixConverter.alfresco(targetCommentNode));
+    }
+
+    @Override
+    public boolean canCreateComment(NodeRef targetNodeRef) {
+        return permissionService.hasPermission(targetNodeRef, IPermissionService.CREATE_CHILDREN);
+    }
+
+    protected Comment toComment(org.alfresco.service.cmr.repository.NodeRef documentNode,
+            org.alfresco.service.cmr.repository.NodeRef commentNodeRef) {
+        NodeRef apixCommentNodeRef = apixConverter.apix(commentNodeRef);
+        NodeMetadata commentMetadata = nodeService.getMetadata(apixCommentNodeRef);
+        String content;
+        try {
+            content = IOUtils.toString(contentService.getContent(apixCommentNodeRef).getInputStream());
+        } catch (IOException ioException) {
+            log.error("Error reading content for comment {}", commentNodeRef, ioException);
+            content = "Comment not available: error reading content";
```

Comment:
Just throw the exception upwards when we can't read comments.
It's a bad idea to replace the comment contents with an error message, because then you can't distinguish between a comment and a comment that failed to be read.